### ISSUE TITLE
Enable static website for storage account

### ISSUE THREAD

**Issue by l3ender** [ansible/ansible_collections_azure#459]
##### SUMMARY
It doesn't appear there is a way to enable the "static website" feature of a storage account.

<img src="https://user-images.githubusercontent.com/2243641/111379394-1532a880-8671-11eb-916f-040b8fb58c92.png" alt="image" width="75%">

##### ISSUE TYPE
- Feature Idea

##### COMPONENT NAME
plugins/modules/azure_rm_storageaccount.py

##### ADDITIONAL INFORMATION
The azure cli equivalent of this is [`az storage blob service-properties`](https://docs.microsoft.com/en-us/cli/azure/storage/blob/service-properties?view=azure-cli-latest). It's non-intuitively named under storage "blob", but these appear to be settings specific for the storage "account".

**Comment by paultaiton** [comment]
I can't commit to this issue as I've already got a lot on my plate currently, however I had some relevant feedback.

The specific tunable is part of the `az storage blob` cli sub-command because in the Python SDK it is part of the azure-storage-blob package ( https://pypi.org/project/azure-storage-blob/ , https://docs.microsoft.com/en-us/azure/developer/python/sdk/storage/storage-blob-readme?view=storage-py-v12 )  and is part of the data plane, not the management plane. The Ansible module that corresponds to this kind of operation is azure_rm_storageblob . The storageblob module does not currently have the ability to tune containers for static websites, ~~however it is definitely the place for it to be added~~.

**Comment by l3ender** [comment]
Thanks for the detail, Paul. I plan to pick this up next week.

How would you suggest modifying azure_rm_storageblob for this feature? The module currently has blob container name as a required field, but modifying service properties can be done on a storage account with no containers.

**Comment by paultaiton** [comment]
storageblob may not be the best place. It's definitely part of the same SDK package, but it may make sense to make a new module, something like azure_rm_storageaccount_staticwebsite  that will instantiate it. I thought the property was container level, not account level.
It could be added to the storageaccount module, however that one is already REALLY complex, so I'd personally go for making a new one specific to staticwebsite. You can re-use the azure-storage-blob from module_utils.

**Comment by l3ender** [comment]
I started looking at implementing this feature (more generically, supporting service properties for a storage account), but I ran into dependency troubles due to the azure-storage package being deprecated and replaced by separate/different components (see [0.37.0 release notes](https://pypi.org/project/azure-storage/0.37.0/)).

Due to this I'd like to wait until dependency packages have been updated (https://github.com/ansible-collections/azure/issues/130). After that I will pick this effort up.

cc @Fred-sun
